NEXT-36303 - In App Purchases docs (#1641)
shopware/docs: guides/plugins/apps/in-app-purchase/index.md

Reviewer: Please update with the newest version:

[In-App Purchases.md](https://github.com/user-attachments/files/18421846/In-App.Purchases.md)

Author: @cyl3x isn't that exactly the base we've worked on in the bus?

Reviewer 2: Almost, there are some small changes regarding plugins

# In-App Purchases

In-App Purchases are a way to lock certain features behind a paywall within the same extension.
This is useful for developers who want to offer a free version of their extension with limited features and a paid version with more features.

> ## How do I receive bought In-App Purchases?

Reviewer 2: shouldn't be a question

Whenever Shopware sends you a request, you'll receive a JWT as a query parameter or in the request body,
depending on whether the request is a GET or POST.
This JWT is signed by our internal systems, ensuring that you, as the app developer, can verify its authenticity and confirm it hasn't been tampered with.

You can use the `shopware/app-php-sdk` for plain PHP or the `shopware/app-bundle` for Symfony to validate and decode the JWT.
An example for plain PHP is available [here](https://github.com/shopware/app-php-sdk/blob/main/examples/index.php).
For Symfony applications, use the appropriate action argument for your route.

### Admin

You will also receive In-App Purchases with the initial `sw-main-hidden` admin request.
To make them accessible, inject them into your JavaScript application.

> Here is an example with an `admin.html.twig` and `shopware/app-bundle`:

Reviewer 2: That sounds a bit plummy, to be honest

```php
#[Route(path: '/app/admin', name: 'admin')]
public function admin(ModuleAction $action): Response {
    return $this->render('admin.html.twig', [
        'inAppPurchases' => $action->inAppPurchases->all(),
    ]);
}
```

```html
<!DOCTYPE html>
<html>
    <head>
        <script>
            try {
                window.inAppPurchases = JSON.parse('{{ inAppPurchases | json_encode | raw }}');
            } catch (e) {
                window.inAppPurchases = [];
                console.error('Unable to decode In-App Purchases', e);
            }
        </script>

        <!-- ... -->
    </head>

    <!-- ... -->
</html>
```

Alternatively you can extract the query parameter from `document.location` on the initial `sw-main-hidden` request,
store it and ask your app-server do properly decode it for you.

> ## Allow the purchase of an In-App Purchases

Reviewer 2: ```suggestion
## Trigger a purchase of an In-App Purchases
```
